### ISSUE TITLE
Support for the new RateLimit standardization proposal

### ISSUE THREAD

**Issue by ioggstream** [nfriedly/express-rate-limit#168]
## I wish

this plugin to support (eg. eventually via configuration) the [new ratelimit standardization proposal](https://tools.ietf.org/id/draft-polli-ratelimit-headers-01.html).

The proposal was:
-  recently presented at the [HTTP Workgroup during the IETF Meeting](http://bit.ly/2qWhwwD)
- implemented by Kong
- under implementation in 3scale

## The proposal

Very similar to your current implementation, but  header names have no `X-` and `Reset` is in `delta-seconds` instead of timestamp. The rationale for the choice is in [FAQ 5](https://ioggstream.github.io/draft-polli-ratelimit-headers/draft-polli-ratelimit-headers.html#rfc.appendix.D).

```
RateLimit-Limit: 60
RateLimit-Remaining: 50
RateLimit-Reset: 5
```

## Proposed change

The line to be changed (even via a configuration parameter) is here:
https://github.com/nfriedly/express-rate-limit/blob/f56434f417180f1c4dbaf2f04a0c150877adce8e/lib/express-rate-limit.js#L83

## Notes

Your feedback on the ratelimit spec is very welcome!

**Comment by nfriedly** [comment]
Hey, that's interesting. I think we can definitely support that with a configuration flag right now. Do you want to put together a pull request with that?

Assuming this gets standardized, I think the next semver major release of rate limit should default to the standard version and perhaps offer a config flag to go back to the current behavior. (Maybe not...)

What is the process of going from a proposal to an official standard?

**Comment by ioggstream** [comment]
I'll work on a PR. Where can I add the configuration parameter?

The standardisation work is in the preliminary phase and we are now gathering implementations.

During the July IETF meeting we'll present the implementors and move forward the spec. 

We can opt for an experimental rfc - with a faster track-or a standard one - slower

**Comment by ioggstream** [comment]
Hi @XVincentX
 seems you're using `express-rate-limit` here: https://github.com/ExpressGateway/express-gateway/blob/8d4e9646e795538ee2abf272b190cabff0ae7568/lib/policies/rate-limit/rate-limit.js 

When this issue will be fixed can we say that `express-gateway` too supports the I-D or we need some more changes there?

**Comment by XVincentX** [comment]
I think I am going to need to update the dependency but other than that, we should be good.

**Comment by nfriedly** [comment]
@ioggstream Apologies, I was out for a few days.

For the PR, you can add a new option here: https://github.com/nfriedly/express-rate-limit/blob/master/lib/express-rate-limit.js#L11

I'm thinking maybe we should name it something like `draft_polli_ratelimit_headers` for now to indicate that it's tied to the draft spec and will change if the spec changes. (And then in the future it'll take over the existing `headers` option.)

And then the implementation around here: https://github.com/nfriedly/express-rate-limit/blob/master/lib/express-rate-limit.js#L76

Does that all sound good to you?

**Comment by ioggstream** [comment]
@sanyamdogra see ^ for your PR :) 

@XVincentX If you don't know him, @sanyamdogra is one of the best GSOC student we had last year.

**Comment by XVincentX** [comment]
Hmm…ok?

**Comment by sanyamdogra** [comment]
Hi @nfriedly @ioggstream @XVincentX, I was in mid of the PR but I guess I'll have to change it accordingly as @nfriedly. I am new with the testing part so I will PR for you all to provide feedback once done.

**Comment by ioggstream** [comment]
@sanyamdogra you can create a draft PR if you like, so we know it's not ready for merge.

**Comment by sanyamdogra** [comment]
Yes, I created the draft PR #169. Kindly have a look and let me know what all have I missed. Thanks!

**Comment by nfriedly** [comment]
Is it valid to omit the `RateLimit-Reset` header? In some cases, e.g. with the memcached store's current implementation, this library does not know when the reset time will be.

**Comment by nfriedly** [comment]
This is now published to npm in v5.1.1
